### PR TITLE
[7.0] Fix ANCM hosting bundle conditions for older OS

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/dotnet/aspnetcore/issues/44113

Reverts most install conditions back to the way they were before arm64 was added in https://github.com/dotnet/aspnetcore/pull/40709 to handle older OS's which do not support NativeMachine, but keeps using NativeMachine for arm64 install condition

## Customer Impact

Without this bug fix, the windows hosting bundle only installs the x86 components for ASP.NET Core for older versions that don't support NativeMachine in wix (i.e. Windows Server 2016).

## Regression?

- [X] Yes
- [ ] No

## Risk

- [ ] High
- [ ] Medium
- [X] Low

## Verification

- [X] Manual (required)

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [X] N/A

### CODE REVIEW

**Comment by snickler** [comment]
I tested with the ci build generated from this PR and it still only installs x64 ANCM and ANCR 

image

**Comment by elinor-fung** [comment]
I expect you need to update the `InstallCondition` for `DotNetRedistLts_x64`
https://github.com/dotnet/aspnetcore/blob/fb537bb78353b74df428b9ec3974c496ee964828/src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs#L39-L43

**Comment by HaoK** [comment]
Ugh thanks @elinor-fung that is probably it

**Comment by HaoK** [comment]
Okay I think this build looks better now @snickler I confirmed on our 2016 VM DotNetRedistLts_x64 and SharedFxRedist_x64 are installed as well now

```
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: WindowsServerHostingBundleOptions.msi, state: Absent, default requested: Present, ba requested: Present, execute: Install, rollback: Uninstall, cache: Yes, uncache: No, dependency: Register
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: AspNetCoreModuleV2_x86, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: No, dependency: None
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: AspNetCoreModuleV2_x64, state: Absent, default requested: Present, ba requested: Present, execute: Install, rollback: Uninstall, cache: Yes, uncache: No, dependency: Register
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: AspNetCoreModuleV2_arm64, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: No, dependency: None
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: DotNetRedistLts_arm64, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: No, dependency: None
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: DotNetRedistLts_x64, state: Absent, default requested: Present, ba requested: Present, execute: Install, rollback: Uninstall, cache: Yes, uncache: No, dependency: None
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: DotNetRedistLts_x86, state: Absent, default requested: Present, ba requested: Present, execute: Install, rollback: Uninstall, cache: Yes, uncache: No, dependency: None
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: SharedFxRedist_arm64, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: No, dependency: None
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: SharedFxRedist_x64, state: Absent, default requested: Present, ba requested: Present, execute: Install, rollback: Uninstall, cache: Yes, uncache: No, dependency: None
[12F8:1130][2022-09-27T17:10:06]i201: Planned package: SharedFxRedist_x86, state: Absent, default requested: Present, ba requested: Present, execute: Install, rollback: Uninstall, cache: Yes, uncache: No, dependency: None
```

**Comment by snickler** [comment]
This is more like it! Complete successful install \o/.

It's good to go from my POV!

image
image
image

File: src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs
Comment (joeloff) [inline src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs:43]: I still think this condition should be `VersionNT64 OR NativeMachine="34404"`. Any non-arm64 result would allow it to install and probably a similar OR condition for the arm64 install condition.

All `VersionNT64` does is tell you that you are on a 64-bit OS, if it's not arm64 it should be amd64, but you could also be on an IA64 platform in which case the x64 installer would not be valid.

**Comment by danmoseley** [comment]
Approved for GA once signed off.

File: src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs
Comment (HaoK) [reply src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs:43]: I guess I'm trying to keep this change to be the smallest possible, prior to the arm64 change this logic was simply
`VersionNT64 AND (NOT OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=&quot;0&quot;)` here in the release/6.0 source https://github.com/dotnet/aspnetcore/blob/release/6.0/src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs#L24

Or did something change in 7.0 where the 6.0 logic here is insufficient (i.e. was IA64 not an issue in 6.0?)

Comment (joeloff) [reply src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs:43]: IA64 is not something we support, it's just that on any unsupported OS, native machine image will always be set to unknown. I'm just paranoid we run into a case where we get a result that's neither unknown nor arm64, but not amd64 and we end up installing the x64 bits. There probably isn't a case like that right now, but the x64 logic now states any machine that's 64-bit and not arm64.

**Review by joeloff — APPROVED** [review]

File: src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs
Comment (HaoK) [reply src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs:43]: Are you comfortable if I make the additional cleanup tweak in 8.0 and keeping the smaller change for 7.0 if IA64 isn't supported?

Comment (HaoK) [reply src/Installers/Windows/WindowsHostingBundle/SharedFramework.wxs:43]: I guess what I'm saying is any issues with the current situation would also show up for 6.0 on x64 too right?

**Comment by snickler** [comment]
@HaoK - Just verified that the GA installer works as expected!

**Pull request merged**
